File: examples/foreign_rust_math/geometry.rs
pub fn circle_area(radius: f64) -> f64 {
    radius * radius * std::f64::consts::PI
}

pub fn rectangle_area(width: f64, height: f64) -> f64 {
    width * height
}

pub fn hypotenuse(a: f64, b: f64) -> f64 {
    (a * a + b * b).sqrt()
}

File: examples/modules/foreign_c_math/geometry.c
#include <stdint.h>
#include <math.h>

double circle_area(double radius) {
    return radius * radius * 3.14159265358979;
}

double rectangle_area(double width, double height) {
    return width * height;
}

double hypotenuse(double a, double b) {
    return sqrt(a * a + b * b);
}

int64_t clamp(int64_t value, int64_t min, int64_t max) {
    if (value < min) return min;
    if (value > max) return max;
    return value;
}
